esx_service/vmdk_ops.py

```diff
@@ -386,12 +390,19 @@ def parse_vol_name(full_vol_name):
     On parse errors raises ValidationError with syntax explanation
     """
     # note: \w in regexp is [a-zA-Z0-9_]
-    groups = re.match("\A([\w\-.]+)@?([\w\-.]+)?$", full_vol_name)
+    groups = re.match(r"\A([a-zA-Z_][\w\_.]+)(@([a-zA-Z_][\w\_\-.]+))?$", full_vol_name)
     if not groups:
-        raise ValidationError("Invalid syntax: '{0}'. " \
-                           "Valid syntax is volume@datastore, where volume or datastore" \
-                           "can contain [a-zA-Z0-9_-.]".format(full_vol_name))
-    return groups.groups()[0], groups.groups()[1]
+        raise ValidationError("Invalid syntax: '{0}'.\n"
```

Comment:
Related to above example, the behavior isn't consistent, 'a.b.c@xxx' works fine, while 'a.b.c@x' doesn't.

>>> s='a.b.c@xxx'
>>> groups = re.match(r"\A([a-zA-Z_][\w\_.]+)(@([a-zA-Z_][\w\_\-.]+))?$", s)
>>> print(groups)
<_sre.SRE_Match object; span=(0, 9), match='a.b.c@xxx'>
>>> s='a.b.c@x'
>>> groups = re.match(r"\A([a-zA-Z_][\w\_.]+)(@([a-zA-Z_][\w\_\-.]+))?$", s)
>>> print(groups)
None